apps/chat/src/pages/api/listing/[...listing].ts

```diff
@@ -27,40 +23,40 @@ import { authOptions } from '@/src/pages/api/auth/[...nextauth]';
 import fetch from 'node-fetch';
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
-  const entityType = getEntityTypeFromPath(req);
-  if (!entityType || !isValidEntityApiType(entityType)) {
-    return res.status(400).json(errorsMessages.notValidEntityType);
-  }
-
   const session = await getServerSession(req, res, authOptions);
   const isSessionValid = validateServerSession(session, req, res);
   if (!isSessionValid) {
     return;
   }
 
   try {
-    const {
-      path = '',
-      filter,
-      bucket,
-      recursive = false,
-    } = req.query as {
-      path: string;
+    const { filter, recursive = false } = req.query as {
       filter?: BackendDataNodeType;
-      bucket: string;
       recursive?: string;
     };
-
     const token = await getToken({ req });
+    const slugs = Array.isArray(req.query.listing)
+      ? req.query.listing
+      : [req.query.listing];
 
-    const url = `${
-      process.env.DIAL_API_HOST
-    }/v1/metadata/${path ? `${encodeApiUrl(path)}` : `${entityType}/${bucket}`}/?limit=1000${recursive ? '&recursive=true' : ''}`;
+    if (!slugs || slugs.length === 0) {
+      throw new OpenAIError(`No path provided`, '', '', '404');
+    }
 
+    const url = `${constructPath(
+      process.env.DIAL_API_HOST,
+      'v1/metadata',
+      encodeSlugs(slugs),
+    )}/?limit=1000${recursive ? '&recursive=true' : ''}`;
```

Comment:
```suggestion
    )}/?limit=1000&recursive=${recursive}`;
```
also maybe it will be better to do something like this:
`const { filter, recursive = false, limit = 1000 } = req.query as {`
and path parameters `?limit=${limit}&recursive=${recursive}`